client/client.go

```diff
@@ -428,10 +428,54 @@ func (c *Client) MakeRestRequest(method string, rpath string, body *container.Co
 	return req, nil
 }
 
+func (c *Client) makeRestRequestRaw(method string, rpath string, payload []byte, authenticated bool) (*http.Request, error) {
```

Comment:
Is there a way to fix the race condition without duplicating the function?
eg. Change `MakeRestRequestRaw` so it calls `makeRestRequestRaw` and `skipLoggingPayload` can be a parameter of `makeRestRequestRaw`. Then `MakeRestRequestRaw` can just pass in `c.skipLoggingPayload` and the Authenticate function can pass in True for skip logging. Would that work?

Reply:
Yes, the next comes removing the repetitive code. I quickly made this PR so that I had the changes saved somewhere.

Reply:
Aha cool, thanks! I'll test it shortly and let you how it goes.